### DISCUSSION TITLE
Is posible use http endpoint in between activities

### DISCUSSION THREAD

**Discussion by nurhidayat-agung** [elsa-workflows/elsa-core#5616] [Q&A]
Hello elsa
Is posible use http endpoint in between activities, that use instance id as parameter in elsa studio

thx

**Comment by glime-ai** [comment]
Yes, it is possible to use an HTTP endpoint in between activities in Elsa Studio, and you can use the instance ID as a parameter. Elsa workflows allow you to create HTTP endpoints that can be triggered externally, and you can pass various parameters, including the workflow instance ID, through query strings, route variables, or headers.

To achieve this, you would typically use the `HttpEndpoint` activity to create an endpoint that can be called via the web. You can configure the path for this endpoint to include route variables, such as the instance ID. For example, you could set up a path like `/process/{instanceId}` where `{instanceId}` is a route variable that represents the workflow instance ID.

When the HTTP endpoint is called with the appropriate instance ID, you can access this value within your workflow and use it in subsequent activities. You can also configure the workflow to handle query string parameters or headers if you prefer to pass the instance ID in those ways.

Remember to set the `CanStartWorkflow` property to `true` if you want the HTTP endpoint to be able to initiate the workflow.

For more detailed information on how to use HTTP activities in Elsa, you can refer to the official documentation provided in the context sources.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
